sections/design/why-spa.tex

```diff
@@ -0,0 +1,48 @@
+\section{Single Page Application or Multi Page Application}
+There are two types of applications that can be created.
+One is a Single Page Application (SPA), the other is a Multi Page application (MPA).
+In this section we will be looking into the two types of architectures and explore the advantages and disadvantages that they have.
+
+\subsection{SPAs}
+What an SPA allows you to do is build an application that simulates that of a desktop application. 
+When a new page is loaded only the necessary content of the previous page is updated.
+JavaScript is often used when developing SPAs but it is often used with some library or framework like React to make the development easier.
+There are also other frameworks such as Angular or Vue which architecture allow for development of dynamic web application just as React does. 
+\\\\
+The main advantages of an SPA is the high speed that comes the unnecessity to load all elements with each new page but only load the changed elements. 
+The frameworks used also makes fast development possible because of the powerful tools that are provided. 
+It is also easy to create a mobile app based on the finished code of an SPA \cite{SPAvsMPAMerehead}.
+SPAs also have greate caching possibilities and the local data can be used offline for when the user has problems with their connection. 
+The pages can then still be shown to the user even with connectivity issues and the local data will be synchronized once the connection is up again \cite{SPAvsMPARuby}.
+\\\\
+Disadvantages of an SPA is the poor optimization for search engines. Most of the pages will not be available to scan because of the way an SPA works. 
```

Comment:
```suggestion
A disadvantage of an SPA is the poor optimization for search engines.
Most of the pages will not be available to scan because of the way an SPA works. 
```